shared/chat/inbox/container.js

```diff
@@ -32,17 +32,46 @@ const passesStringFilter = (filter: string, toCheck: string): boolean => {
   return toCheck.toLowerCase().indexOf(filter.toLowerCase()) >= 0
 }
 
-const passesParticipantFilter = (participants: I.List<string>, filter: string, you: ?string): boolean => {
+const passesParticipantFilter = (filter: string, participants: Array<string>, you: ?string): boolean => {
   if (!filter) {
     return true
   }
 
   // don't filter you out if its just a convo with you!
-  const justYou = participants.count() === 1 && participants.first() === you
+  const justYou = participants.length === 1 && participants[0] === you
   const names = justYou ? participants : participants.filter(p => p !== you)
   return names.some(n => passesStringFilter(filter, n))
 }
 
+// Simple score for a filter. returns 1 for exact match. 0.75 for full name match
+// in a group conversation. 0.5 for a partial match
+// 0 for no match
+function scoreFilter(filter, stringToFilterOn, you) {
+  if (stringToFilterOn.indexOf(',') > 0) {
```

Comment:
we join then immediately split?

Reply:
yeah, I was deciding between having this function take either an array and a string or just an array. but I think you're right. I'll change!